File: java/1662_CheckIfTwoStringArraysAreEquivalent.java
class Solution {
    public boolean arrayStringsAreEqual(String[] word1, String[] word2) {
        int idx1 = 0, idx2 = 0;
        int a = 0, b = 0;
        while (idx1 < word1.length && idx2 < word2.length) {
            while (a < word1[idx1].length() && b < word2[idx2].length()) {
                if (word1[idx1].charAt(a++) != word2[idx2].charAt(b++)) return false;
            }
            if (a >= word1[idx1].length()) {
                idx1++;
                a = 0;
            }
            if (b >= word2[idx2].length()) {
                idx2++;
                b = 0;
            }
        }
        return idx1 == word1.length && idx2 == word2.length;
    }
}

File: python/1662_CheckIfTwoStringArraysAreEquivalent.py
class Solution:
    def arrayStringsAreEqual(self, word1: List[str], word2: List[str]) -> bool:
        idx1 = idx2 = a = b = 0
        N, M = len(word1), len(word2)
        while idx1 < N and idx2 < M:
            while a < len(word1[idx1]) and b < len(word2[idx2]):
                if word1[idx1][a] != word2[idx2][b]:
                    return False
                a += 1
                b += 1
            if a >= len(word1[idx1]):
                idx1 += 1
                a = 0
            if b >= len(word2[idx2]):
                idx2 += 1
                b = 0
        
        return idx1 == N and idx2 == M
